File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/rust_solutions/1544-make-the-string-great.rs
impl Solution {
    pub fn make_good(s: String) -> String {
        let mut stack: Vec<char> = Vec::new();
        for c in s.chars() {
            if let Some(&top) = stack.last() {
                if (c.is_lowercase() && top.is_uppercase() && c.to_uppercase().next().unwrap() == top) ||
                   (c.is_uppercase() && top.is_lowercase() && c.to_lowercase().next().unwrap() == top) {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            } else {
                stack.push(c);
            }
        }
        stack.iter().collect()
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/go_solutions/1544-make-the-string-great.go
func makeGood(s string) string {
	stack := []rune{}
	for _, r := range s {
		if len(stack) > 0 && (stack[len(stack)-1] == r+32 || stack[len(stack)-1] == r-32) {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, r)
		}
	}
	return string(stack)
}